Source/monster.cpp

```diff
@@ -4212,9 +4172,13 @@ void GolumAi(int i)
 				enemy.position.last = golem.position.tile;
 				for (int j = 0; j < 5; j++) {
 					for (int k = 0; k < 5; k++) {
-						int enemyId = dMonster[golem.position.tile.x + k - 2][golem.position.tile.y + j - 2]; // BUGFIX: Check if indexes are between 0 and 112
+						int golemXIndex = golem.position.tile.x + k - 2;
+						int golemYIndex = golem.position.tile.y + j - 2;
+						if (golemXIndex < 0 || golemXIndex > 112 || golemYIndex < 0 || golemYIndex > 112)

```

Comment:
this will let 112 through = out of bounds
and we have InDungeonBounds function for that